backend/cmd/server/servicemanager.go

```diff
@@ -45,12 +46,20 @@ import (
 	"github.com/asgardeo/thunder/internal/userschema"
 )
 
+// observabilitySvc is the observability service instance. This is used for graceful shutdown.
+var observabilitySvc observability.ObservabilityServiceInterface
+
 // registerServices registers all the services with the provided HTTP multiplexer.
-func registerServices(mux *http.ServeMux, jwtService jwt.JWTServiceInterface) {
+func registerServices(
+	mux *http.ServeMux,
+	jwtService jwt.JWTServiceInterface,
+) {
 	logger := log.GetLogger()
 
+	observabilitySvc := observability.Initialize()
```

Comment:
This will not work as intended. 

Reason: 
The := operator creates a new local variable inside the function, which shadows the package-level variable.
So the global observabilitySvc is never assigned, and remains nil.
```suggestion
	observabilitySvc = observability.Initialize()
```